src/compile/header/parse.ts

```diff
@@ -1,16 +1,17 @@
-import {AxisOrient} from 'vega';
+import {AxisOrient, SignalRef} from 'vega';
 import {isArray} from 'vega-util';
 import {FacetChannel, FACET_CHANNELS} from '../../channel';
 import {title as fieldDefTitle} from '../../channeldef';
 import {contains, getFirstDefined} from '../../util';
+import {isSignalRef} from '../../vega.schema';
 import {assembleAxis} from '../axis/assemble';
 import {FacetModel} from '../facet';
 import {parseGuideResolve} from '../resolve';
 import {getHeaderProperty} from './common';
 import {HeaderChannel, HeaderComponent} from './component';
 
-export function getHeaderType(orient: AxisOrient) {
-  if (orient === 'top' || orient === 'left') {
+export function getHeaderType(orient: AxisOrient | SignalRef) {
+  if (orient === 'top' || orient === 'left' || isSignalRef(orient)) {
```

Comment:
Why do we default to `header` if orient is a signal?

Reply:
Because header / footer can't be signal in Vega.  We should add some comments here.